src/cls/CMakeLists.txt

```diff
@@ -48,7 +48,7 @@ if (WITH_RBD)
 
   set(cls_rbd_client_srcs rbd/cls_rbd_client.cc rbd/cls_rbd_types.cc)
   add_library(cls_rbd_client STATIC ${cls_rbd_client_srcs})
-  target_link_libraries(cls_rbd_client cls_lock_client)
+  target_link_libraries(cls_rbd_client librados libneorados)
```

Comment:
Dropping `cls_lock_client` makes sense to me, but the addition of `librados` and `libneorados` seems to make this target inconsistent with all other `*_client` targets in `src/cls/CMakeLists.txt` -- both ones that are used by RBD (`cls_lock_client` and `cls_journal_client`) and ones that aren't.  Most of them interface with RADOS in a very similar fashion but not even one has it as a dependency.